embedded/ahtree/ahtree.go

```diff
@@ -142,8 +145,13 @@ func Open(path string, opts *Options) (*AHtree, error) {
 }
 
 func OpenWith(pLog, dLog, cLog appendable.Appendable, opts *Options) (*AHtree, error) {
-	if !validOptions(opts) || pLog == nil || dLog == nil || cLog == nil {
-		return nil, ErrIllegalArguments
+	if pLog == nil || dLog == nil || cLog == nil {
+		return nil, fmt.Errorf("%w: invalid appendables", ErrIllegalArguments)
```

Comment:
may be you would want to apply the same format of error message here as used below for nil options?

```
		return fmt.Errorf("%w: nil options", ErrInvalidOptions)
		return fmt.Errorf("%w: nil appendable", ErrInvalidOptions) (something similar)

```